terminus-ui/chip/src/chip-collection.component.ts

```diff
@@ -0,0 +1,588 @@
+
+  /**
+   * Whether or not this chip collection is selectable. When a chip collection is not selectable,
+   * all the chips are not selectable.
+   */
+  @Input()
+  public set isSelectable(value: boolean) {
+    this._selectable = coerceBooleanProperty(value);
+    if (this.chips) {
+      this.chips.forEach(chip => {
+        chip.chipCollectionSelectable = this._selectable;
+      });
+    }
+  }
+  public get isSelectable(): boolean {
+    return this._selectable;
+  }
+  protected _selectable = true;
+
+  @Input()
+  public orientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  @Input()
+  public set tabIndex(value: number) {
+    this._userTabIndex = value;
+    this._tabIndex = value;
+  }
+  public get tabIndex(): number {
+    return this._tabIndex;
+  }
+
+  /**
+   * Orientation of the chip collection.
+   */
+  @Input('aria-orientation')
+  public ariaOrientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Event emitted when the chip collection value has been changed by the user. */
+  @Output()
+  public readonly collectionChange = new EventEmitter<TsChipCollectionChange>();
+
+  /**
+   * Emitted when a chip is to be removed.
+   */
+  @Output()
+  public readonly removed = new EventEmitter<TsChipEvent>();
+
+  /**
+   * Emitted when tab pressed with chip focused
+   */
+  @Output()
+  public readonly tabUpdateFocus = new EventEmitter();
+
+  constructor(protected elementRef: ElementRef<HTMLElement>,
+              private changeDetectorRef: ChangeDetectorRef,
+              public zone: NgZone,
+              @Optional() private dir: Directionality,) {
+    zone.runOutsideAngular(() => {
+      this._role = this.empty ? null : 'listbox';
+    });
+  }
+
+  public ngOnInit() {
+    this.selectionModel = new SelectionModel<TsChipComponent>(this.allowMultipleSelections, undefined, false);
+  }
+
+  public ngAfterViewInit() {
+    this.keyManager = new FocusKeyManager<TsChipComponent>(this.chips)
+      .withWrap()
+      .withVerticalOrientation()
+      .withHorizontalOrientation(this.dir ? this.dir.value : 'ltr');
+
+    this.keyManager.tabOut.pipe(untilComponentDestroyed(this)).subscribe(() => {
+      this.tabUpdateFocus.emit();
+    });
+
+    // When the collection changes, re-subscribe
+    this.chips.changes.pipe(startWith(null), untilComponentDestroyed(this)).subscribe(() => {
+      if (this.isDisabled || this.isReadonly) {
+        // Since this happens after the content has been
+        // checked, we need to defer it to the next tick.
+        Promise.resolve().then(() => {
+          this.syncChipsState();
+        });
+      }
+
+      this.resetChips();
+
+      // Check to see if we need to update our tab index
+      Promise.resolve().then(() => {
+        this.updateTabIndex();
+      });
+
+      // Check to see if we have a destroyed chip and need to refocus
+      this.updateFocusForDestroyedChips();
+      this.propagateChanges();
+
+    });
+  }
+
+  public ngOnDestroy() { }
```

Comment:
We should add a comment that this empty function is needed for `untilComponentDestroyed`